src/main/kotlin/no/nav/syfo/identhendelse/IdenthendelseService.kt

```diff
@@ -0,0 +1,63 @@
+package no.nav.syfo.identhendelse
+
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.client.pdl.PdlClient
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.identhendelse.database.*
+import no.nav.syfo.identhendelse.kafka.COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES
+import no.nav.syfo.identhendelse.kafka.KafkaIdenthendelseDTO
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class IdenthendelseService(
+    private val database: DatabaseInterface,
+    private val pdlClient: PdlClient,
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(IdenthendelseService::class.java)
+
+    fun handleIdenthendelse(identhendelse: KafkaIdenthendelseDTO) {
+        if (identhendelse.folkeregisterIdenter.size > 1) {
+            val activeIdent = identhendelse.getActivePersonident()
+            if (activeIdent != null) {
+                val inactiveIdenter = identhendelse.getInactivePersonidenter()
+                val numberOfUpdatedIdenter = updateAllTables(activeIdent, inactiveIdenter)
+                if (numberOfUpdatedIdenter > 0) {
+                    log.info("Identhendelse: Updated $numberOfUpdatedIdenter rows based on Identhendelse from PDL")
+                    COUNT_KAFKA_CONSUMER_PDL_AKTOR_UPDATES.increment(numberOfUpdatedIdenter.toDouble())
+                }
+            } else {
+                log.warn("Mangler gyldig ident fra PDL")
```

Comment:
Kanskje inkludere noe om "Identhendelse ignored: Mangler gyldig.."? (så man skjønner hva det dreier seg om)

Reply:
Selve loggeren er vel knyttet til klassen via `LoggerFactory.getLogger(IdenthendelseService::class.java)`. Men jeg satt litt og tenkte på om det burde blitt kastet en feil her. For den kommer hit dersom det ikke er noen identer i PDL som er aktive, og det er jo på en måte en feil 🤔